static/src/javascripts/projects/common/modules/ui/selection-sharing.spec.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+
+import $ from 'lib/$';
+import {
+    init as selectionSharingInit,
+    updateSelection,
+} from './selection-sharing';
+
+const markup = `<div class="js-article__body"><p>Some text to select</p></div>`;
+
+jest.mock('lib/detect', () => ({
+    hasTouchScreen: () => false,
+}));
+
+jest.mock('rangefix', () => ({
+    getBoundingClientRect: () => ({ left: 0, bottom: 0 }),
+}));
+
+beforeEach(() => {
+    if (document.body) {
+        document.body.innerHTML = markup;
+    }
+});
+
+afterEach(() => {
+    if (document.body) {
+        document.body.innerHTML = '';
+    }
+});
+
+test('should be initially hidden in the article body', () => {
+    selectionSharingInit();
+
+    expect($('.selection-sharing').length).toBe(1);
+    expect($('.selection-sharing').hasClass('selection-sharing--active')).toBe(
+        false
+    );
+});
+
+test('should be visible when text is selected in the article body', () => {
+    // required mocks (not present in jsdom at time of writing)
+    class MockRange {
```

Comment:
All of this and the subsequent $FlowFixMe feel hacky so thoughts welcome.

Reply:
It's very sad, but a great workaround. I really hope they stop this jsdom nonsense at some point, as the implementation is super incomplete ...